Fix LR scheduler issue on CI

### Describe the bug

Currently CI is failing on segmenation example:

- https://github.com/pytorch-ignite/code-generator/actions/runs/4466293918/jobs/7844298360

```
[ignite]: Configuration: 
{'accumulation_steps': 4,
 'backend': None,
 'data_path': '/home/runner/data',
 'debug': False,
 'eval_batch_size': 2,
 'eval_epoch_length': 4,
 'filename_prefix': 'training',
 'limit_sec': 60,
 'log_every_iters': 2,
 'lr': 0.007,
 'max_epochs': 2,
 'n_saved': 2,
 'num_classes': 21,
 'num_workers': 2,
 'output_dir': PosixPath('logs/20230320-090125-backend-None-lr-0.007'),
 'patience': 2,
 'save_every_iters': 2,
 'seed': 666,
 'train_batch_size': 2,
 'train_epoch_length': 4,
 'use_amp': False}
Traceback (most recent call last):
  File "/home/runner/work/code-generator/code-generator/dist-tests/vision-segmentation-all/main.py", line 181, in <module>
    main()
  File "/home/runner/work/code-generator/code-generator/dist-tests/vision-segmentation-all/main.py", line 177, in main
    p.run(run, config=config)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/ignite/distributed/launcher.py", line 316, in run
    func(local_rank, *args, **kwargs)
  File "/home/runner/work/code-generator/code-generator/dist-tests/vision-segmentation-all/main.py", line 83, in run
    trainer.add_event_handler(Events.ITERATION_STARTED, lr_scheduler)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/ignite/engine/engine.py", line 319, in add_event_handler
    _check_signature(handler, "handler", self, *(event_args + args), **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/ignite/engine/utils.py", line 10, in _check_signature
    signature = inspect.signature(fn)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/inspect.py", line 3113, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/inspect.py", line 2862, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/inspect.py", line 2261, in _signature_from_callable
    raise TypeError('{!r} is not a callable object'.format(obj))
TypeError: <torch.optim.lr_scheduler.LambdaLR object at 0x7fdc0baab0a0> is not a callable object
Error: Process completed with exit code 1.
```

--- a/src/templates/template-text-classification/main.py
+++ b/src/templates/template-text-classification/main.py
@@ -11,9 +11,13 @@ from ignite.metrics import Accuracy, Loss
 from ignite.utils import manual_seed
 from models import TransformerModel
 from torch import nn, optim
-from torch.optim.lr_scheduler import _LRScheduler
 from trainers import setup_evaluator, setup_trainer
 from utils import *
+
+try:
+    from torch.optim.lr_scheduler import _LRScheduler as PyTorchLRScheduler
+except ImportError:
+    from torch.optim.lr_scheduler import LRScheduler as PyTorchLRScheduler
 
 os.environ["TOKENIZERS_PARALLELISM"] = "false"  # remove tokenizer paralleism warning
 
@@ -78,10 +82,10 @@ def run(local_rank: int, config: Any):
     (config.output_dir / "config-lock.yaml").write_text(yaml.dump(config))
     trainer.logger = evaluator.logger = logger
 
-    if isinstance(lr_scheduler, _LRScheduler):
+    if isinstance(lr_scheduler, PyTorchLRScheduler):
         trainer.add_event_handler(
             Events.ITERATION_COMPLETED,
-            lambda engine: cast(_LRScheduler, lr_scheduler).step(),
+            lambda engine: cast(PyTorchLRScheduler, lr_scheduler).step(),
         )
     elif isinstance(lr_scheduler, LRScheduler):
         trainer.add_event_handler(Events.ITERATION_COMPLETED, lr_scheduler)

--- a/src/templates/template-vision-segmentation/main.py
+++ b/src/templates/template-vision-segmentation/main.py
@@ -21,6 +21,7 @@ try:
 except ImportError:
     from torch.optim.lr_scheduler import _LRScheduler as PyTorchLRScheduler
 
+
 def run(local_rank: int, config: Any):
     # make a certain seed
     rank = idist.get_rank()

--- a/src/templates/template-vision-segmentation/main.py
+++ b/src/templates/template-vision-segmentation/main.py
@@ -11,11 +11,15 @@ from ignite.metrics import ConfusionMatrix, IoU, mIoU
 from ignite.utils import manual_seed
 from models import setup_model
 from torch import nn, optim
-from torch.optim.lr_scheduler import _LRScheduler, LambdaLR
+from torch.optim.lr_scheduler import LambdaLR
 from trainers import setup_evaluator, setup_trainer
 from utils import *
 from vis import predictions_gt_images_handler
 
+try:
+    from torch.optim.lr_scheduler import LRScheduler as PyTorchLRScheduler
+except ImportError:
+    from torch.optim.lr_scheduler import _LRScheduler as PyTorchLRScheduler
 
 def run(local_rank: int, config: Any):
     # make a certain seed
@@ -71,10 +75,10 @@ def run(local_rank: int, config: Any):
     (config.output_dir / "config-lock.yaml").write_text(yaml.dump(config))
     trainer.logger = evaluator.logger = logger
 
-    if isinstance(lr_scheduler, _LRScheduler):
+    if isinstance(lr_scheduler, PyTorchLRScheduler):
         trainer.add_event_handler(
             Events.ITERATION_COMPLETED,
-            lambda engine: cast(_LRScheduler, lr_scheduler).step(),
+            lambda engine: cast(PyTorchLRScheduler, lr_scheduler).step(),
         )
     elif isinstance(lr_scheduler, LRScheduler):
         trainer.add_event_handler(Events.ITERATION_COMPLETED, lr_scheduler)

